### ISSUE TITLE
openSpineItemElementCfi(idref, cfi) opens the page containing the beginning of the paragraph instead of the page showing the cfi.

### ISSUE THREAD

**Issue by Twibit** [readium/readium-shared-js#121]
To create annotation, I use ```ReadiumSDK.reader.getCurrentSelectionCfi()```.
To open the page containing the annotation I use ```ReadiumSDK.reader.openSpineItemElementCfi(idref, cfi)```.

When I select some text from a paragraphe that begin in the previous page and then call the following code,  it opens the page containing the begining of the paragraphe instead of the page containing the selected text.
```javascript
var idref = JSON.parse(ReadiumSDK.reader.bookmarkCurrentPage()).idref;
var cfi = ReadiumSDK.reader.getCurrentSelectionCfi().cfi;
ReadiumSDK.reader.openSpineItemElementCfi(idref, cfi);
```

**Comment by Twibit** [comment]
The problem is that the navigation logic doesn't use the cfi's range to find the best position.

**Comment by danielweck** [comment]
I believe @JCCR is working on that (see feature branch below), am I right?
https://github.com/readium/readium-shared-js/tree/feature/improve_cfi_accuracy

**Comment by danielweck** [comment]
Is this issue a duplicate of this older one?
https://github.com/readium/readium-shared-js/issues/35

**Comment by danielweck** [comment]
I think this is the same problem: https://github.com/readium/readium-shared-js/issues/157
We discussed this bug on today's conference call. Juan (see feature branch mentioned above) and I will talk more about the ill-designed `/4/2/6@0,y` "bookmark" CFI (save + restore reading location based on visual 2D coordinates in the reflowable document)

**Comment by danielweck** [comment]
Is that fixed by Juan's PR? ( @JCCR )
https://github.com/readium/readium-shared-js/pull/210

**Comment by mooshee** [comment]
This is blocked by #257 at least on iOS.

**Comment by hajarghaem** [comment]
Hi @danielweck, @JCCR,
Could you please let me know if this issue is fixed? It does not work for me?
Thanks in advance,
Kind regards,
-Hajar
